woocommerce/woocommerce-ios
# Product Subscriptions: Analytics

Closes: #11183


This PR adds the following analytics:

1. `product_list_loaded` event now includes `is_eligible_for_subscriptions` property.
2. Expiration events:
  - `product_details_view_subscription_expiration_date_tapped` for simple subscription products
  - `product_variation_view_subscription_expiration_date_tapped`  for variation subscription products
3. Free trial setting events:
  - `product_details_view_subscription_free_trial_tapped` for simple subscription products
  - `product_variation_view_subscription_free_trial_tapped`  for variation subscription products
4. Done tapping events: 
  - `product_subscription_expiration_done_button_tapped` event when "Done" button on Expiration setting is tapped, with `has_changed_data` property
  - `product_subscription_free_trial_done_button_tapped` event when "Done" button on Free Trial setting is tapped, with `has_changed_data` property
 
## Testing instructions


1. Have a test site containing both simple and variation subscription products,
2. In the app, open products list screen. Once products are loaded, check for event `product_list_loaded` with `is_eligible_for_subscriptions` set as true,
3. Open a simple subscription product,
4. Open the "Free trial" setting. Check for event `product_details_view_subscription_free_trial_tapped`
5. Change some settings there and tap Done. Check for event `product_subscription_free_trial_done_button_tapped` with `has_changed_data` property set as true.
6. Open the "Expire after" setting. Check for event `product_subscription_free_trial_done_button_tapped`. 
7. Change some settings there and tap Done. Check for event `product_subscription_expiration_done_button_tapped` with `has_changed_data` property set as true.
8. Open a variation subscription product,
9. Open the "Free trial" setting. Check for event `product_variation_view_subscription_free_trial_tapped`
10. Change some settings there and tap Done. Check for event `product_subscription_free_trial_done_button_tapped` with `has_changed_data` property set as true.
11. Open the "Expire after" setting. Check for event `product_variation_view_subscription_expiration_date_tapped`. 
12. Change some settings there and tap Done. Check for event `product_subscription_expiration_done_button_tapped` with `has_changed_data` property set as true.
